core/src/main/java/com/linecorp/armeria/client/EventLoopAcquisitionIndex.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+final class EventLoopAcquisitionIndex {
+    private final int sizeEventLoopPool;
+    private final int maxNumEventLoops;
+    private final int acquisitionStartIndex;
+    private int nextUnusedEventLoopOffset;
+
+    EventLoopAcquisitionIndex(int acquisitionStartIndex, int maxNumEventLoops, int sizeEventLoopPool) {
+        this.acquisitionStartIndex = acquisitionStartIndex;
+        this.maxNumEventLoops = maxNumEventLoops;
+        this.sizeEventLoopPool = sizeEventLoopPool;
+        nextUnusedEventLoopOffset = ThreadLocalRandom.current().nextInt(maxNumEventLoops);
+    }
+
+    int nextAcquirableIndex() {
+        return (acquisitionStartIndex + nextUnusedEventLoopOffset) % sizeEventLoopPool;
+    }
```

Comment:
What do you think of just moving the offset after acquiring an index?

We can also make `moveOffset` private, resulting in a cleaner API design

```suggestion
    int nextAcquirableIndex() {
        final int index = (acquisitionStartIndex + nextUnusedEventLoopOffset) % sizeEventLoopPool;
        moveOffset();
        return index;
    }
```